### DISCUSSION TITLE
[Question][Hw6] Can I add more modules(files)?

### DISCUSSION THREAD

**Discussion by bongjunj** [prosyslab-classroom/cs348-information-security#479] [Q&A]
Name: 장봉준

I find it somewhat limited to implement a fuzzer in just `coverage.ml`.
I want to implement more complex structure using OCaml's functor.
To do so, adding more modules in separate files to the codebase is preferable.

However, the document says I can't change the directory structure.
But, still, I'm not going to edit the given code.
Can I do this?

I made some changes to the `dune` build script. And everything is working OK.

**Comment by yeonhee-ryou** [comment]
You can define multiple modules in a single file so you can define your own module in the src/campaign.ml.

We require you not to fix the directory structure for grading and plagiarism check.
Even if there is no problem with gradescope auto grade, please do not to change it.
